src/libcollections/vec.rs

```diff
@@ -1039,18 +1039,24 @@ impl<T> Vec<T> {
     #[inline]
     #[stable(feature = "append", since = "1.4.0")]
     pub fn append(&mut self, other: &mut Self) {
-        self.reserve(other.len());
-        let len = self.len();
-        unsafe {
-            ptr::copy_nonoverlapping(other.as_ptr(), self.get_unchecked_mut(len), other.len());
-        }
-
-        self.len += other.len();
         unsafe {
+            let count = other.len();
+            self.append_elements(other.as_ptr(), count);
             other.set_len(0);
         }
     }
 
+    /// Appends a elements to `Self`, without moving.
```

Comment:
Updated type to `*const [T]`.

I've changed to docs to the suggested, but the original made more sense to me. It's only half moving the elements. It has directly copied them, but the original buffer has not been modified, so if the caller doesn't do something, they'll have copied potentially uncopiable data, and have data unsafety. 🤷‍♂️